pkg/authorizer/token_retriever_wi.go

```diff
@@ -0,0 +1,48 @@
+package authorizer
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/confidential"
+)
+
+// Get auth token from service account token
+func AcquireARMTokenFromServiceAccountToken(ctx context.Context, tenantID, clientID string) (types.AccessToken, error) {
+
+	const (
+		authorityHost = "https://login.microsoftonline.com/"
+		resource      = "https://management.azure.com/.default"
+	)
+
+	cred := confidential.NewCredFromAssertionCallback(func(context.Context, confidential.AssertionRequestOptions) (string, error) {
+		return readJWTFromFS()
+	})
+
+	confidentialClientApp, err := confidential.New(
+		clientID,
+		cred,
+		confidential.WithAuthority(fmt.Sprintf("%s%s/oauth2/token", authorityHost, tenantID)))
+	if err != nil {
+		return "", fmt.Errorf("Unable to get new confidential client app: %w", err)
+	}
+
+	authResult, err := confidentialClientApp.AcquireTokenByCredential(ctx, []string{"/.default"})
```

Comment:
You need to pass `resource` to `AcquireTokenByCredential`. Right now you're passing `/.default`